spec/controllers/uploads_spec.rb

```diff
@@ -0,0 +1,46 @@
+require "spec_helper"
+
+RSpec.describe UploadsController do
+  render_views
+
+  describe "GET index" do
+    it "assigns @uploads" do
+      user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).
+                                                  and_return(user)
+      upload = create(:upload)
+      get :index
+      puts("there are this many uploads #{Upload.count}")
+      puts("there are this many uploads with meaning #{Upload.with_meaning.count}")
+      expect(assigns(:uploads)).to eq([upload])
+    end
+
+    it "renders the index template" do
+      user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).
+                                                  and_return(user)
+      subject { get :index }
+    end
+
+    it "uploads the file" do
+      user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).
+                                                  and_return(user)
+
+      get :create, { params: { audio: file_to_upload } }
+      expect(response).to redirect_to(new_upload_path)
+    end
+
+    def file_to_upload
+      ActionDispatch::Http::UploadedFile.new({
+        content_type: "audio/wav",
```

Comment:
Use 2 spaces for indentation in a hash, relative to the first position after the preceding left parenthesis.